check/src/unify_type.rs

```diff
@@ -133,11 +133,8 @@ impl<I> Substitutable for ArcType<I> {
     type Variable = TypeVariable;
 
     fn new(id: u32) -> ArcType<I> {
-        Type::variable(TypeVariable::new(id))
-    }
-
-    fn from_variable(var: TypeVariable) -> ArcType<I> {
-        Type::variable(var)
+        // FIXME: using `Kind::hole()` here results in test failures
+        Type::variable(TypeVariable::new(id, Kind::typ()))
```

Comment:
I am fairly certain the correct fix is to make `new_var` take a kind as well. Looked the uses of the function over and it looks like about ~90 % of its uses then `Kind::Type` is correct but there are a few where it should be `Kind::Row` or propagated from a generic variable.